python/vowpalwabbit/pyvw.py

```diff
@@ -533,7 +533,7 @@ def __init__(self, label, cost=0., partial_prediction=0., wap_value=0.):
             self.costs.append(wc)
 
     def __str__(self):
-        return ' '.join([str(c.label) + ':' + str(c.cost) for c in self.costs])
+        return ' '.join(["{}:{}:{}".format(c.action, c.cost, c.wap_value) for c in self.costs])
```

Comment:
CSOAA labels are not formed like this. You can read the parse function to get an idea of how these work. https://github.com/VowpalWabbit/vowpal_wabbit/blob/3c0f0ffb84c47701521abba6450a8cf7b39fba83/vowpalwabbit/cost_sensitive.cc#L124